### DISCUSSION TITLE
Custom scopes for oauth login?

### DISCUSSION THREAD

**Discussion by StephenOTT** [microsoft/vscode-powerquery-sdk#304] [Q&A]
Is there a way to set user arguments for the oauth login? 
I would like to allow the user to supply custom scopes for the login process.

**Comment by mattmasson** [comment]
The `StartLogin` function receives the data source path (credential path), so as long as the arguments are[ included in your data source path](https://learn.microsoft.com/en-us/power-query/handling-resource-path) you should be able to factor them into your oauth flow.

That said, I can't think of another connector that we've shipped that does this. Typically the connectors will know exactly what scopes they need to access a given data source.
